x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/filter/multi_select_filter.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiSelectableOption } from '@elastic/eui';
+import {
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSelectable,
+  EuiSpacer,
+  EuiText,
+  EuiTextColor,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { isEqual } from 'lodash/fp';
+import React, { useEffect, useState } from 'react';
+import * as i18n from './translations';
+
+type FilterOption<T extends string, K extends string = string> = EuiSelectableOption<{
+  key: K;
+  label: T;
+}>;
+
+export type { FilterOption as MultiSelectFilterOption };
+
+export const mapToMultiSelectOption = <T extends string>(
+  options: T[],
+  labelsMap?: Record<string, string>
```

Comment:
This should be type `Record<string, T>`, though I'm not sure why we're doing `T extends string` instead of just `string` with no generics.

Reply:
I removed all generics from multi_select_filter